File: 1-1.hs
import Data.Char
import System.IO

firstDigit :: String -> Char
firstDigit []     = '0'
firstDigit (x:xs) = if isDigit x then x else firstDigit xs

lastDigit :: String -> Char
lastDigit s = firstDigit (reverse s)

calculateLine :: String -> Int
calculateLine s = read [firstDigit(s), lastDigit(s)]

calculateCalibration :: [String] -> Int
calculateCalibration []   = 0
calculateCalibration (l:ls) = calculateLine l + calculateCalibration ls


main :: IO ()
main =  do
    putStr "Result: "
    x <- getContents
    print(calculateCalibration (lines x))

File: 1-1.rs
use std::io;

fn first_digit(line: &str) -> u32
{
    for c in line.chars() {
        if c.is_digit(10) {
            return c.to_digit(10).unwrap();
        }
    }

    return 0;
}

fn last_digit(line: &str) -> u32
{
    return first_digit(line.chars().rev().collect::<String>().as_str());
}

fn calculate_line(line: &str) -> u32
{
    return (first_digit(line) * 10) + last_digit(line);
}

fn main() {
    let lines = io::stdin().lines();
    let mut calibrations = Vec::new();

    for line in lines {
        calibrations.push(calculate_line(line.unwrap().as_str()));
    }

    println!("{}", calibrations.into_iter().reduce(|acc, e| acc + e).unwrap());
}